<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introducing mRNA bias into simulations with scaling factors • SimBu</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introducing mRNA bias into simulations with scaling factors">
<meta property="og:description" content="SimBu">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SimBu</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/simulator_documentation.html">Getting started with the pseudo-bulk simulator R package</a>
    </li>
    <li>
      <a href="../articles/simulator_scaling_factors.html">Introducing mRNA bias into simulations with scaling factors</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="simulator_scaling_factors_files/header-attrs-2.8/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introducing mRNA bias into simulations with scaling factors</h1>
                        <h4 class="author">Alexander Dietrich</h4>
            
            <h4 class="date">11/29/2021</h4>
      
      
      <div class="hidden name"><code>simulator_scaling_factors.Rmd</code></div>

    </div>

    
    
<div id="using-scaling-factors" class="section level1">
<h1 class="hasAnchor">
<a href="#using-scaling-factors" class="anchor"></a>Using scaling factors</h1>
<p>This package allows the user to introduce an mRNA bias into pseudo-bulk RNA-seq samples. Different cell-types contain different amounts of mRNA (Dendritic cells for examples contain much more than Neutrophils); this bias can be added into simulations artificially in different ways.<br>
The scaling factors will always be applied on the single-cell dataset first, altering its expression profiles accordingly, and then the pseudo-bulk samples are generated with an aggregation method (<code>mean</code> by default). By default we <code>multiply</code> the scaling factor with the dataset, but you can change this to <code>division</code> in the <code>scaling_factor_aggregation</code> parameter of <code><a href="../reference/simulate_bulk.html">simulate_bulk()</a></code>.</p>
<div id="pre-defined-scaling-factors" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-defined-scaling-factors" class="anchor"></a>Pre-defined scaling factors</h2>
<p>Some studies have proposed scaling factors for immune cells, such as EPIC <span class="citation">(<a href="#ref-Racle2017" role="doc-biblioref">Racle et al. 2017</a>)</span> or quantiseq <span class="citation">(<a href="#ref-Finotello2019" role="doc-biblioref">Finotello et al. 2019</a>)</span>, deconvolution tools which correct for the mRNA bias internally using these values:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">epic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B cells"</span>, 
           <span class="st">"Macrophages"</span>,
           <span class="st">"Monocytes"</span>, 
           <span class="st">"Neutrophils"</span>,
           <span class="st">"NK cells"</span>,
           <span class="st">"T cells"</span>,
           <span class="st">"T cells CD4"</span>,
           <span class="st">"T cells CD8"</span>,
           <span class="st">"T helper cells"</span>,
           <span class="st">"T regulatory cells"</span>,
           <span class="st">"otherCells"</span>,
           <span class="st">"default"</span><span class="op">)</span>,
  mRNA <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.4016</span>,
           <span class="fl">1.4196</span>,
           <span class="fl">1.4196</span>,
           <span class="fl">0.1300</span>,
           <span class="fl">0.4396</span>,
           <span class="fl">0.3952</span>,
           <span class="fl">0.3952</span>,
           <span class="fl">0.3952</span>,
           <span class="fl">0.3952</span>,
           <span class="fl">0.3952</span>,
           <span class="fl">0.4000</span>,
           <span class="fl">0.4000</span><span class="op">)</span>
  
<span class="op">)</span>

<span class="va">epic</span>
<span class="co">#&gt;                  type   mRNA</span>
<span class="co">#&gt; 1             B cells 0.4016</span>
<span class="co">#&gt; 2         Macrophages 1.4196</span>
<span class="co">#&gt; 3           Monocytes 1.4196</span>
<span class="co">#&gt; 4         Neutrophils 0.1300</span>
<span class="co">#&gt; 5            NK cells 0.4396</span>
<span class="co">#&gt; 6             T cells 0.3952</span>
<span class="co">#&gt; 7         T cells CD4 0.3952</span>
<span class="co">#&gt; 8         T cells CD8 0.3952</span>
<span class="co">#&gt; 9      T helper cells 0.3952</span>
<span class="co">#&gt; 10 T regulatory cells 0.3952</span>
<span class="co">#&gt; 11         otherCells 0.4000</span>
<span class="co">#&gt; 12            default 0.4000</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">quantiseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B cells"</span>,
           <span class="st">"Macrophages"</span>,
           <span class="st">"MacrophagesM2"</span>,
           <span class="st">"Monocytes"</span>,
           <span class="st">"Neutrophils"</span>,
           <span class="st">"NK cells"</span>,
           <span class="st">"T cells CD4"</span>,
           <span class="st">"T cells CD8"</span>,
           <span class="st">"T regulatory cells"</span>,
           <span class="st">"Dendritic cells"</span>,
           <span class="st">"T cells"</span><span class="op">)</span>,
  mRNA <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> <span class="fl">65.66148</span>,
            <span class="fl">138.11520</span>,
            <span class="fl">119.35447</span>,
            <span class="fl">130.65455</span>,
            <span class="fl">27.73634</span>,
            <span class="fl">117.71584</span>,
            <span class="fl">63.87200</span>,
            <span class="fl">70.25659</span>,
            <span class="fl">72.55110</span>,
            <span class="fl">140.76091</span>,
            <span class="fl">68.89323</span><span class="op">)</span>
  
<span class="op">)</span>

<span class="va">quantiseq</span>
<span class="co">#&gt;                  type      mRNA</span>
<span class="co">#&gt; 1             B cells  65.66148</span>
<span class="co">#&gt; 2         Macrophages 138.11520</span>
<span class="co">#&gt; 3       MacrophagesM2 119.35447</span>
<span class="co">#&gt; 4           Monocytes 130.65455</span>
<span class="co">#&gt; 5         Neutrophils  27.73634</span>
<span class="co">#&gt; 6            NK cells 117.71584</span>
<span class="co">#&gt; 7         T cells CD4  63.87200</span>
<span class="co">#&gt; 8         T cells CD8  70.25659</span>
<span class="co">#&gt; 9  T regulatory cells  72.55110</span>
<span class="co">#&gt; 10    Dendritic cells 140.76091</span>
<span class="co">#&gt; 11            T cells  68.89323</span></code></pre></div>
<p>When you want to apply one of these scaling factors into your simulation (therefore in-/decreasing the expression signals for the cell-types), use the <code>custom</code> value for the <code>scaling_factor</code> parameter. Note, that the input for <code>custom_scaling_vector</code> needs to be a named list, so we need to transform the dataframes from above slightly:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scaling_epic</span> <span class="op">&lt;-</span> <span class="va">epic</span><span class="op">$</span><span class="va">mRNA</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">scaling_epic</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">epic</span><span class="op">$</span><span class="va">type</span>
<span class="va">sim_epic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">dataset</span>,
                          scenario <span class="op">=</span> <span class="st">"random"</span>,
                          scaling_factor<span class="op">=</span><span class="st">"custom"</span>, 
                          custom_scaling_vector <span class="op">=</span> <span class="va">scaling_epic</span>, 
                          nsamples <span class="op">=</span> <span class="fl">100</span>, 
                          ncells<span class="op">=</span><span class="fl">1000</span>,
                          ncores<span class="op">=</span><span class="fl">4</span><span class="op">)</span></code></pre></div>
<p>With this approach, we can also try out some custom scaling factors, for example increasing the expression levels for a single cell-type (B cells) by 10-fold compared to the rest. All cell-types which are not mentioned in the named list given to <code>custom_scaling_vector</code> will be transformed with a scaling factor of 1, meaning nothing changes for them.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sim_extreme_b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dataset</span>, 
                               scenario <span class="op">=</span> <span class="st">"random"</span>,
                               scaling_factor <span class="op">=</span> <span class="st">"custom"</span>,
                               custom_scaling_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B cells"</span><span class="op">=</span><span class="fl">10</span><span class="op">)</span>, 
                               nsamples <span class="op">=</span> <span class="fl">100</span>, 
                               ncells <span class="op">=</span> <span class="fl">1000</span>, 
                               ncores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<p><strong>Important:</strong> Take care, that the cell-type annotation names in your dataset are the same as in the scaling factor! Otherwise the scaling factor will not be applied or even worse, applied to a different cell-type.</p>
</div>
<div id="dataset-specific-scaling-factors" class="section level2">
<h2 class="hasAnchor">
<a href="#dataset-specific-scaling-factors" class="anchor"></a>Dataset specific scaling factors</h2>
<p>You can also choose to calculate scaling factors, which are depending on your provided single-cell dataset. Compared to the previous section, this will give a unique value for each cell rather than a cell-type, making it possibly more sensitive.</p>
<div id="reads-and-genes" class="section level3">
<h3 class="hasAnchor">
<a href="#reads-and-genes" class="anchor"></a>Reads and genes</h3>
<p>Two straight forward approaches would be the number of reads or number of expressed genes/features. For this, we need to generate our initial dataset object a little bit differently and add more information to the internal annotation table:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">  <span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dataset.html">dataset</a></span><span class="op">(</span>annotation <span class="op">=</span> <span class="va">annotation</span>,
                count_matrix <span class="op">=</span> <span class="va">count_matrix</span>,
                name <span class="op">=</span> <span class="st">"reads_and_genes"</span>,
                read_number_col <span class="op">=</span> <span class="st">"total_reads"</span>,      <span class="co"># add read number column</span>
                additional_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gene_number"</span><span class="op">)</span><span class="op">)</span>   <span class="co"># add gene number column</span></code></pre></div>
<p>Here our <code>annotation</code> dataframe contains additional columns such as <code>total_reads</code> and <code>gene_number</code>. The first gives the number of reads per cell, the other the number of expressed genes per cell. The number of reads will be stored in the dataset annotation in the <code>read_number</code> column, all columns we add via the <code>additional_cols</code> parameter will keep their original name. We can now use these values for mRNA bias introduction based on reads or genes:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">sim_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">ds</span>, 
                           scenario <span class="op">=</span> <span class="st">"random"</span>, 
                           scaling_factor<span class="op">=</span><span class="st">"read-number"</span>,  <span class="co"># use number of reads as scaling factor</span>
                           nsamples <span class="op">=</span> <span class="fl">100</span>, 
                           ncells<span class="op">=</span><span class="fl">1000</span>, 
                           ncores<span class="op">=</span><span class="fl">4</span><span class="op">)</span>

<span class="va">sim_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ds</span>,
                           scenario <span class="op">=</span> <span class="st">"random"</span>,
                           scaling_factor <span class="op">=</span> <span class="st">"gene_number"</span>, <span class="co"># use gene_number column as scaling factor</span>
                           nsamples <span class="op">=</span> <span class="fl">100</span>,
                           ncells <span class="op">=</span> <span class="fl">1000</span>,
                           ncores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<p>These options would also allow you to use other numerical measurements you have for the single cells as scaling factors, such as weight or size for example.</p>
</div>
<div id="spike-ins" class="section level3">
<h3 class="hasAnchor">
<a href="#spike-ins" class="anchor"></a>Spike-ins</h3>
<p>One other numerical measurement can be spike-ins. Usually the number of reads mapped to spike-in molecules per cell is given in the cell annotations. If this is the case, they can be stored in the dataset annotation using the <code>spike_in_col</code> parameter, where you indicate the name of the column from the <code>annotation</code> dataframe in which the spike-in information is stored. To calculate a scaling factor from this, the number of reads are also necessary, so we will add this information as well (as above using the <code>read_number_col</code> parameter).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R">  <span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dataset.html">dataset</a></span><span class="op">(</span>annotation <span class="op">=</span> <span class="va">annotation</span>,
                count_matrix <span class="op">=</span> <span class="va">count_matrix</span>,
                name <span class="op">=</span> <span class="st">"spike-ins"</span>,
                spike_in_col <span class="op">=</span> <span class="st">"ercc_sum"</span>,   <span class="co"># add spike-in column</span>
                read_number_col <span class="op">=</span> <span class="st">"total_reads"</span><span class="op">)</span></code></pre></div>
<p>The scaling factor with spike-ins is calculated as the “% of reads NOT mapped to spike-in reads,” or: <code>(n_reads - n_spike_in)/n_reads</code> for each cell. We apply it like this:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sim_spike</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ds</span>,
                           scenario <span class="op">=</span> <span class="st">"random"</span>,
                           scaling_factor <span class="op">=</span> <span class="st">"spike-in"</span>, <span class="co"># use spike-in scaling factor</span>
                           nsamples <span class="op">=</span> <span class="fl">100</span>,
                           ncells <span class="op">=</span> <span class="fl">1000</span>,
                           ncores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="census---estimate-mrna-counts-per-cell" class="section level2">
<h2 class="hasAnchor">
<a href="#census---estimate-mrna-counts-per-cell" class="anchor"></a>Census - estimate mRNA counts per cell</h2>
<p><a href="http://cole-trapnell-lab.github.io/monocle-release/docs/#census">Census</a> is an approach which tries to convert TPM counts into relative transcript counts. This basically means, you get the mRNA counts per cell, which can differ between cell-types.<br><span class="citation">(<a href="#ref-Qiu2017" role="doc-biblioref">Qiu et al. 2017</a>)</span> state in their paper, that it should only be applied to TPM/FPKM normalized data, but I tried it out with raw expression counts as well, which worked as well.<br>
Census calculates a vector with a scaling value for each cell in a sample. You can switch this feature on, by setting the <code>scaling_factor</code> parameter to <code>census</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sim_census</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dataset_file</span>,
                             scenario <span class="op">=</span> <span class="st">"spike-in"</span>, 
                             scaling_factor <span class="op">=</span> <span class="st">"census"</span>,   <span class="co"># use census scaling factor</span>
                             nsamples <span class="op">=</span> <span class="fl">100</span>, 
                             ncells<span class="op">=</span><span class="fl">1000</span>, 
                             ncores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<p>In our analysis we found, that Census is basically a complicated way of estimating the number of expressed genes per cell. It will remain to the user to decide if he/she wants to use census or simply the number of expressed genes (as shown above) as scaling factor.</p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Finotello2019" class="csl-entry">
Finotello, Francesca, Clemens Mayer, Christina Plattner, Gerhard Laschober, Dietmar Rieder, Hubert Hackl, Anne Krogsdam, et al. 2019. <span>“Molecular and Pharmacological Modulators of the Tumor Immune Contexture Revealed by Deconvolution of RNA-Seq Data.”</span> <em>Genome Medicine</em> 11 (1): 34. <a href="https://doi.org/10.1186/s13073-019-0638-6">https://doi.org/10.1186/s13073-019-0638-6</a>.
</div>
<div id="ref-Qiu2017" class="csl-entry">
Qiu, Xiaojie, Andrew Hill, Jonathan Packer, Dejun Lin, Yi-An Ma, and Cole Trapnell. 2017. <span>“Single-Cell mRNA Quantification and Differential Analysis with Census.”</span> <em>Nature Methods</em> 14 (3): 309–15. <a href="https://doi.org/10.1038/nmeth.4150">https://doi.org/10.1038/nmeth.4150</a>.
</div>
<div id="ref-Racle2017" class="csl-entry">
Racle, Julien, Kaat de Jonge, Petra Baumgaertner, Daniel E. Speiser, and David Gfeller. 2017. <span>“Simultaneous Enumeration of Cancer and Immune Cell Types from Bulk Tumor Gene Expression Data.”</span> <em>eLife</em> 6 (November): e26476. <a href="https://doi.org/10.7554/eLife.26476">https://doi.org/10.7554/eLife.26476</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alexander Dietrich.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
