<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started with the pseudo-bulk simulator R package • SimBu</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with the pseudo-bulk simulator R package">
<meta property="og:description" content="SimBu">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SimBu</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/simulator_documentation.html">Getting started with the pseudo-bulk simulator R package</a>
    </li>
    <li>
      <a href="../articles/simulator_scaling_factors.html">Introducing mRNA bias into simulations with scaling factors</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="simulator_documentation_files/header-attrs-2.8/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting started with the pseudo-bulk simulator R package</h1>
                        <h4 class="author">Alexander Dietrich</h4>
            
      
      
      <div class="hidden name"><code>simulator_documentation.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install the package, run this: (Since it is a private repository, you currently need an authentication token for this repository.)</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"omnideconv/simulator"</span><span class="op">)</span> <span class="co">#add auth_token=XXX to download from private repository</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SimBu</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com">data.table</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://Matrix.R-forge.R-project.org/">Matrix</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Matrix'</span>
<span class="co">#&gt; The following objects are masked from 'package:tidyr':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand, pack, unpack</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat">Seurat</a></span><span class="op">)</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">RColorBrewer</span><span class="op">)</span></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Cell-type deconvolution is an important analysis-step while processing bulk RNA-seq experiments and many tools exist to tackle this issue. Though the user lacks a good comparison between those tools and when one might be more beneficial than others.<br>
The goal of  is to create benchmark datasets, where pseudo bulk samples with predefined cell-type fractions can be simulated using expression data from single-cell RNAseq experiments.</p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p>This chapter covers all you need to know to quickly simulate some pseudo-bulk samples!</p>
<div id="setup-the-simulator" class="section level2">
<h2 class="hasAnchor">
<a href="#setup-the-simulator" class="anchor"></a>Setup the simulator</h2>
<p>This package can simulate samples from local or public data. We will start with the public data integration, since it requires mostly no manual download of datasets. As a public database, sfaira <span class="citation">(<a href="#ref-Fischer2020" role="doc-biblioref">Fischer et al. 2020</a>)</span> is used, which is a dataset and model repository for single-cell RNA-sequencing data. It gives access to about 233 datasets from human and mouse with more than 3 million cells in total. You can browse them interactively here: <a href="https://theislab.github.io/sfaira-portal/Datasets">https://theislab.github.io/sfaira-portal/Datasets</a>. Note that only annotated datasets will be downloaded!<br>
In order to use this database, we first need to install it. We recommend to use <a href="https://docs.conda.io/projects/conda/en/latest/index.html">conda</a> to create a new environment and install sfaira into this directory. This ensures that no other packages interfere with the installation.<br><em>Note:</em> currently the simulator seems to work only with the develop version of sfaira.<br>
A recommended step by step installation would look like this:</p>
<pre><code>conda create --name sfaira python=3.9 pip
conda activate sfaira
/path/to/conda/envs/sfaira/bin/pip install git+https://github.com/theislab/sfaira.git@dev #this ensures that the correct pip is used</code></pre>
<p>Watch out that you are using pip inside of the new environment, not the global pip!<br></p>
<p>You do not need to activate the new conda environment each time in order to use it in the R package.<br>
To setup sfaira inside of the R package use this command:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">setup_list</span> <span class="op">&lt;-</span> <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/setup_sfaira.html">setup_sfaira</a></span><span class="op">(</span>python_path <span class="op">=</span> <span class="st">"/nfs/home/students/adietrich/.conda/envs/sfaira/bin/python3"</span>,
                                      env_name <span class="op">=</span> <span class="st">"sfaira"</span>, 
                                      basedir <span class="op">=</span> <span class="st">"/nfs/home/students/adietrich/ma/data/sfaira"</span><span class="op">)</span></code></pre></div>
<p>The <code>python_path</code> parameter is used to set the path to the python executable <strong>inside</strong> of the newly created conda environment. Similarly, the <code>env_name</code> parameter is the name of this newly created conda environment. These two parameters need to be set correctly, so that sfaira can be used later on. You can find out the python-path in a conda environment named <code>sfaira</code> like this:<br></p>
<pre><code>activate sfaira
which python3</code></pre>
<p>The third parameter describes the name of a directory in which the raw files, meta data and cached data from downloading datasets from sfaira are stored into.<br>
You need to save the output of the command above in a variable for later use.<br>
If you do not wish to use sfaira, just skip this step entirely.</p>
</div>
<div id="creating-a-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-a-dataset" class="anchor"></a>Creating a dataset</h2>
<p>The simulator package works with two internally defined data-structures: <em>datasets</em> and <em>databases</em>.<br>
We will now create a dataset of samples from human pancreas using the <code>organisms</code> and <code>tissues</code> parameter. You can provide a single word (like we do here) or for example a list of tissues you want to download: <code><a href="https://rdrr.io/r/base/c.html">c("pancreas","lung")</a></code>. An additional parameter is the <code>assays</code> parameter, where you subset the database further to only download datasets from certain sequencing assays (for examples <code>Smart-seq2</code>).<br>
The <code>name</code> parameter is used later on to give each sample a unique name.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ds_human_pancreas</span> <span class="op">&lt;-</span> <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/dataset_sfaira_multiple.html">dataset_sfaira_multiple</a></span><span class="op">(</span>sfaira_setup <span class="op">=</span> <span class="va">setup_list</span>,
                                                        organisms <span class="op">=</span> <span class="st">"human"</span>, 
                                                        tissues <span class="op">=</span> <span class="st">"pancreas"</span>, 
                                                        name<span class="op">=</span><span class="st">"human_pancreas"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Removing datasets without cell-type annotation..."</span>
<span class="co">#&gt; [1] "Downloading datasets..."</span>
<span class="co">#&gt; [1] "Streamlining features &amp; meta-data..."</span>
<span class="co">#&gt; [1] "Using rownames for cell-IDs."</span>
<span class="co">#&gt; [1] "Filtering genes..."</span>
<span class="co">#&gt; [1] "Created dataset."</span></code></pre></div>
<p>Currently there are two datasets in sfaira from human pancreas, which have cell-type annotation. The package will download them for you automatically and merge them together into a single expression matrix and a streamlined annotation table, which we can use for our simulation.<br>
It can happen, that some datasets from sfaira are not (yet) ready for the automatic download, an error message will then appear in R, telling you which file to download and where to put it.<br></p>
<p>If you wish to see all datasets which are included in sfaira you can use the following command:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_datasets</span> <span class="op">&lt;-</span> <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/sfaira_overview.html">sfaira_overview</a></span><span class="op">(</span>setup_list <span class="op">=</span> <span class="va">setup_list</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">all_datasets</span><span class="op">)</span>
<span class="co">#&gt;                                                                                          id</span>
<span class="co">#&gt; 1:                            human_liver_2019_10x3v2_popescu_001_10.1038/s41586-019-1652-y</span>
<span class="co">#&gt; 2:                 human_lungparenchyma_2019_10x3v2_madissoon_001_10.1186/s13059-019-1906-x</span>
<span class="co">#&gt; 3:                      human_esophagus_2019_10x3v2_madissoon_002_10.1186/s13059-019-1906-x</span>
<span class="co">#&gt; 4:                         human_spleen_2019_10x3v2_madissoon_003_10.1186/s13059-019-1906-x</span>
<span class="co">#&gt; 5:                              human_lung_2019_dropseq_braga_001_10.1038/s41591-019-0468-5</span>
<span class="co">#&gt; 6: human_lungparenchyma_2019_10x3transcriptionprofiling_braga_001_10.1038/s41591-019-0468-5</span>
<span class="co">#&gt;       author                       doi annotated normalization</span>
<span class="co">#&gt; 1:   Popescu 10.1038/s41586-019-1652-y      TRUE           raw</span>
<span class="co">#&gt; 2: Madissoon 10.1186/s13059-019-1906-x      TRUE           raw</span>
<span class="co">#&gt; 3: Madissoon 10.1186/s13059-019-1906-x      TRUE           raw</span>
<span class="co">#&gt; 4: Madissoon 10.1186/s13059-019-1906-x      TRUE           raw</span>
<span class="co">#&gt; 5:     Braga 10.1038/s41591-019-0468-5      TRUE           raw</span>
<span class="co">#&gt; 6:     Braga 10.1038/s41591-019-0468-5      TRUE        scaled</span>
<span class="co">#&gt;                             assay           organ organism</span>
<span class="co">#&gt; 1:                      10x 3' v2           liver    human</span>
<span class="co">#&gt; 2:                      10x 3' v2 lung parenchyma    human</span>
<span class="co">#&gt; 3:                      10x 3' v2       esophagus    human</span>
<span class="co">#&gt; 4:                      10x 3' v2          spleen    human</span>
<span class="co">#&gt; 5:                       Drop-seq            lung    human</span>
<span class="co">#&gt; 6: 10x 3' transcription profiling lung parenchyma    human</span></code></pre></div>
</div>
<div id="simulate-pseudo-bulk-datasets" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-pseudo-bulk-datasets" class="anchor"></a>Simulate pseudo bulk datasets</h2>
<p>We are now ready to simulate the first pseudo bulk samples with the created dataset:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">simulation</span> <span class="op">&lt;-</span> <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ds_human_pancreas</span>,
                                       scenario <span class="op">=</span> <span class="st">"random"</span>, 
                                       scaling_factor <span class="op">=</span> <span class="st">"NONE"</span>, 
                                       ncells<span class="op">=</span><span class="fl">1000</span>, 
                                       nsamples <span class="op">=</span> <span class="fl">50</span>, 
                                       ncores <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>
<span class="co">#&gt; Finished simulation.</span></code></pre></div>
<p><code>ncells</code> sets the number of cells in each sample, while <code>nsamples</code> sets the total amount of simulated samples.<br>
You can also control the number of cells per sample by using the <code>total_read_counts</code> parameter, which sets the total number of counts per sample. This can be used to simulate different sequencing depths. If you wish to use this parameter, it is important to know what type of expression data you have (raw or normalized). If you for example use TPMs as expression values and set the total read counts to <code>1e7</code>, this will get you really huge samples with many cells, since TPM values are generally smaller than the raw count values.</p>
<p>Currently there are 6 <code>scenarios</code> implemented in the package:<br></p>
<ul>
<li><p><em>random</em>: for each cell in a sample, the cell-type is chosen randomly. This scenario is depending on the background distribution of the cell-types in the created dataset. If for example your dataset contains 20% B-cells, 40% T-cells and 40% Macrophages, this scenario will create samples where this distribution is mirrored closely.<br></p></li>
<li><p><em>uniform</em>: this creates samples, where all existing cell-types in the dataset appear in the same proportions. So using a dataset with 3 cell-types, this will simulate samples, where all cell-type fractions are 1/3.<br></p></li>
<li><p><em>spike-in</em>: here you need to set two additional parameters for the <code><a href="../reference/simulate_bulk.html">simulate_bulk()</a></code> function: <code>spike_in_cell_type</code> sets the cell-type you want to be over-representing and <code>spike_in_amount</code> sets the fraction of this cell-type. You could for example use <code>B-cell</code> and <code>0.5</code> to create samples, where 50% are B-cells and the rest is filled randomly with other cell-types.<br></p></li>
<li><p><em>spill-over</em>: this simulates samples with only a single cell-type present. Like with <em>spike-in</em> you need to set this cell-type with the additional parameter <code>spillover_cell_type</code>.<br></p></li>
<li><p><em>unique</em>: this creates simulations of only one single cell-type. You have to provide the name of this cell-type with the <code>unique_cell_type</code> parameter.</p></li>
<li><p><em>custom</em>: here you are able to create your own set of cell-type fractions. When using this scenario, you additionally need to provide a dataframe in the <code>custom_scenario_data</code> parameter, where each row represents one sample (therefore the number of rows need to match the <code>nsamples</code> parameter). Each column has to represent one cell-type, which also occurs in the dataset and describes the fraction of this cell-type in a sample. The fractions per sample need to sum up to 1. An example can be seen here:</p></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">unique_scenario_dataframe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>
  <span class="st">"B cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.1</span>, <span class="fl">0.5</span>, <span class="fl">0.3</span><span class="op">)</span>,
  <span class="st">"T cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.8</span>, <span class="fl">0.2</span>, <span class="fl">0.5</span><span class="op">)</span>,
  <span class="st">"NK cells"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.1</span>, <span class="fl">0.3</span>, <span class="fl">0.2</span><span class="op">)</span>,
  row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample1"</span>,<span class="st">"sample2"</span>,<span class="st">"sample3"</span>,<span class="st">"sample4"</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">unique_scenario_dataframe</span>
<span class="co">#&gt;         B.cells T.cells NK.cells</span>
<span class="co">#&gt; sample1     0.2     0.3      0.5</span>
<span class="co">#&gt; sample2     0.1     0.8      0.1</span>
<span class="co">#&gt; sample3     0.5     0.2      0.3</span>
<span class="co">#&gt; sample4     0.3     0.5      0.2</span></code></pre></div>
</div>
<div id="results" class="section level2">
<h2 class="hasAnchor">
<a href="#results" class="anchor"></a>Results</h2>
<p>The <code>simulation</code> object contains three named entries:<br></p>
<ul>
<li><p><code>pseudo_bulk_raw</code>: a sparse matrix with - in this case - 50 columns and about 60k rows (this is the number of features/genes). This is the simulated expression matrix.<br></p></li>
<li><p><code>pseudo_bulk</code>: same as above, but CPM normalized samples.<br></p></li>
<li><p><code>cell_fractions</code>: a dataframe with 50 rows and 15 columns (this is the number of cell-types). It gives the fractions for each cell-type in each sample.<br></p></li>
<li><p><code>expression_set</code>: a Biobase <a href="https://rdrr.io/bioc/Biobase/man/class.ExpressionSet.html">ExpressionSet</a><br></p></li>
</ul>
<p>Finally here is a boxplot of the resulting simulation:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">simulation</span><span class="op">$</span><span class="va">cell_fractions</span>
<span class="va">fractions</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">)</span>
<span class="va">frac_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/gather.html">gather</a></span><span class="op">(</span><span class="va">fractions</span>, <span class="va">cell_type</span>, <span class="va">fraction</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">frac_long</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">fraction</span>, y<span class="op">=</span><span class="va">sample</span>, fill<span class="op">=</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Cell-type fractions for 50 pseudo-bulk RNAseq samples"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="va">frac_long</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="simulator_documentation_files/figure-html/unnamed-chunk-8-1.png" width="768"></p>
</div>
</div>
<div id="more-features" class="section level1">
<h1 class="hasAnchor">
<a href="#more-features" class="anchor"></a>More features</h1>
<div id="using-your-own-data" class="section level2">
<h2 class="hasAnchor">
<a href="#using-your-own-data" class="anchor"></a>Using your own data</h2>
<p>The simulator also has the option to create datasets with your own scRNAseq experiments. You can create datasets in three different ways:<br></p>
<ul>
<li>Using a expression matrix &amp; annotation file</li>
</ul>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># change paths accordingly</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/Matrix.html">Matrix</a></span><span class="op">(</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/as.matrix.html">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"~/ma/data/Maynard/X_tpm.csv"</span><span class="op">)</span><span class="op">)</span>, sparse <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"~/ma/data/Maynard/var.csv"</span><span class="op">)</span>
<span class="va">cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"~/ma/data/Maynard/obs.csv"</span><span class="op">)</span>
<span class="va">cellnames</span> <span class="op">&lt;-</span> <span class="va">cells</span><span class="op">$</span><span class="va">Run</span>
<span class="va">genenames</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">$</span><span class="va">symbol</span>
<span class="fu"><a href="https://rdrr.io/r/base/dimnames.html">dimnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">cellnames</span>, <span class="va">genenames</span><span class="op">)</span>
<span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>

<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"~/ma/data/Maynard/annotation_obs.csv"</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Run"</span>,<span class="st">"cell_type"</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ID"</span>, <span class="st">"cell_type"</span><span class="op">)</span>

<span class="va">dataset_file</span> <span class="op">&lt;-</span> <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/dataset.html">dataset</a></span><span class="op">(</span>annotation <span class="op">=</span> <span class="va">annotation</span>,
                                   count_matrix <span class="op">=</span> <span class="va">counts</span>,
                                   name<span class="op">=</span><span class="st">"Maynard"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Filtering genes..."</span>
<span class="co">#&gt; [1] "Created dataset."</span></code></pre></div>
<ul>
<li>Using a h5ad expression data &amp; annotation file</li>
</ul>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># change paths accordingly</span>
<span class="va">h5ad_file</span> <span class="op">&lt;-</span> <span class="st">"~/ma/data/Travaglini/Travaglini_Krasnow_2020_Lung_SS2.h5ad"</span>
<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">fread</a></span><span class="op">(</span><span class="st">"~/ma/data/Travaglini/obs_extended.csv"</span><span class="op">)</span>
<span class="co"># renaming the columns</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op">==</span> <span class="st">"free_annotation"</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="st">"cell_type"</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op">==</span> <span class="st">"index"</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="st">"ID"</span>

<span class="va">dataset_h5ad</span> <span class="op">&lt;-</span> <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/dataset_h5ad.html">dataset_h5ad</a></span><span class="op">(</span>annotation <span class="op">=</span> <span class="va">annotation</span>,
                                        h5ad_file <span class="op">=</span> <span class="va">h5ad_file</span>, 
                                        name <span class="op">=</span> <span class="st">"Travaglini"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Filtering genes..."</span>
<span class="co">#&gt; [1] "Created dataset."</span></code></pre></div>
<ul>
<li>Using a Seurat Object and annotation file</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># download Seurat object from tabula muris</span>
<span class="co"># https://figshare.com/ndownloader/files/13088531</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="st">"~/ma/data/tabula_muris/tabula_muris_bladder.Robj"</span><span class="op">)</span>
<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/UpdateSeuratObject.html">UpdateSeuratObject</a></span><span class="op">(</span><span class="va">tiss</span><span class="op">)</span>
<span class="co">#&gt; Updating from v2.X to v3.X</span>
<span class="co">#&gt; Validating object structure</span>
<span class="co">#&gt; Updating object slots</span>
<span class="co">#&gt; Ensuring keys are in the proper strucutre</span>
<span class="co">#&gt; Ensuring feature names don't have underscores or pipes</span>
<span class="co">#&gt; Object representation is consistent with the most current Seurat version</span>
<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span>
<span class="co"># renaming the columns</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span> <span class="op">==</span> <span class="st">"free_annotation"</span><span class="op">)</span><span class="op">]</span><span class="op">&lt;-</span><span class="st">"cell_type"</span>
<span class="va">annotation</span><span class="op">$</span><span class="va">ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">annotation</span><span class="op">)</span>

<span class="va">dataset_seurat</span> <span class="op">&lt;-</span> <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/dataset_seurat.html">dataset_seurat</a></span><span class="op">(</span>annotation <span class="op">=</span> <span class="va">annotation</span>,
                                            seurat_obj <span class="op">=</span> <span class="va">seurat_obj</span>,
                                            name<span class="op">=</span><span class="st">"TabulaMuris_Bladder"</span><span class="op">)</span>
<span class="co">#&gt; [1] "Filtering genes..."</span>
<span class="co">#&gt; [1] "Created dataset."</span></code></pre></div>
<p><strong>Important:</strong> the annotation dataframe you are feeding into the dataset, needs these two columns: <code>ID</code> for the cell ID, which has to be in common with the cell IDs in the expression matrix and <code>cell_type</code>, which gives the cell-type for a cell.<br></p>
</div>
<div id="simulate-using-a-whitelist-and-blacklist-of-cell-types" class="section level2">
<h2 class="hasAnchor">
<a href="#simulate-using-a-whitelist-and-blacklist-of-cell-types" class="anchor"></a>Simulate using a whitelist (and blacklist) of cell-types</h2>
<p>Sometimes, you are only interested in specific cell-types, but the dataset you are using has way too many cell-types:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="va">dataset_file</span><span class="op">@</span><span class="va">annotation</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Monocyte conventional"      "Macrophage"                </span>
<span class="co">#&gt;  [3] "Endothelial cell"           "Mast cell"                 </span>
<span class="co">#&gt;  [5] "B cell"                     "T cell CD8"                </span>
<span class="co">#&gt;  [7] "T cell CD4"                 "Alevolar cell type 2"      </span>
<span class="co">#&gt;  [9] "Basal"                      "Smooth muscle cell"        </span>
<span class="co">#&gt; [11] "NK cell"                    "other C17"                 </span>
<span class="co">#&gt; [13] "Monocyte non-conventional"  "cDC2"                      </span>
<span class="co">#&gt; [15] "Epithelial cell other C5"   "Fibroblast adventitial"    </span>
<span class="co">#&gt; [17] "Club"                       "Epithelial cell other C4"  </span>
<span class="co">#&gt; [19] "DC mature"                  "Epithelial cell other C1"  </span>
<span class="co">#&gt; [21] "Plasma cell"                "Epithelial cell other C3"  </span>
<span class="co">#&gt; [23] "Pericyte"                   "Alevolar cell type 1"      </span>
<span class="co">#&gt; [25] "T cell regulatory"          "pDC"                       </span>
<span class="co">#&gt; [27] "Epithelial cell other C2"   "Ciliated"                  </span>
<span class="co">#&gt; [29] "other C21"                  "other C19"                 </span>
<span class="co">#&gt; [31] "Fibroblast alevolar"        "cDC1"                      </span>
<span class="co">#&gt; [33] "Mesothelial"                "Endothelial cell lymphatic"</span>
<span class="co">#&gt; [35] "DC Lagerhans"               "Epithelial cell dividing"  </span>
<span class="co">#&gt; [37] "T cell dividing"            "other C29"                 </span>
<span class="co">#&gt; [39] "Ionocyte"                   "other C32"                 </span>
<span class="co">#&gt; [41] "other C33"                  "Goblet"</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="va">dataset_file</span><span class="op">@</span><span class="va">annotation</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 42</span></code></pre></div>
<p>You can handle this issue during simulation using the <code>whitelist</code> parameter:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">simulation</span> <span class="op">&lt;-</span>  <span class="fu">simulator</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/simulator/man/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">dataset_file</span>,
                                        scenario <span class="op">=</span> <span class="st">"spike_in"</span>, 
                                        scaling_factor <span class="op">=</span> <span class="st">"NONE"</span>, 
                                        spike_in_cell_type <span class="op">=</span> <span class="st">"NK cell"</span>,
                                        spike_in_amount <span class="op">=</span> <span class="fl">0.5</span>,
                                        ncells<span class="op">=</span><span class="fl">1000</span>, 
                                        nsamples <span class="op">=</span> <span class="fl">50</span>, 
                                        ncores <span class="op">=</span> <span class="fl">12</span>,
                                        whitelist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NK cell"</span>,<span class="st">"B cell"</span>,<span class="st">"T cell CD8"</span>, <span class="st">"T cell CD4"</span>, <span class="st">"Macrophage"</span>, <span class="st">"Monocyte conventional"</span>, <span class="st">"Monocyte non-conventional"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Finished simulation.</span>

<span class="va">fractions</span> <span class="op">&lt;-</span> <span class="va">simulation</span><span class="op">$</span><span class="va">cell_fractions</span>
<span class="va">fractions</span><span class="op">$</span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">)</span>
<span class="va">frac_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/gather.html">gather</a></span><span class="op">(</span><span class="va">fractions</span>, <span class="va">cell_type</span>, <span class="va">fraction</span>, <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fractions</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">frac_long</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">fraction</span>, y<span class="op">=</span><span class="va">sample</span>, fill<span class="op">=</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="st">"Cell-type fractions for 50 pseudo-bulk RNAseq samples"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html">brewer.pal</a></span><span class="op">(</span><span class="fl">8</span>, <span class="st">"Set2"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/duplicated.html">unique</a></span><span class="op">(</span><span class="va">frac_long</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="simulator_documentation_files/figure-html/unnamed-chunk-13-1.png" width="768"></p>
<p>In the same way, you can also provide a <code>blacklist</code> parameter, where you name the cell-types you <strong>don’t</strong> want to be included in your simulation.</p>
</div>
<div id="filtering-and-aggregation" class="section level2">
<h2 class="hasAnchor">
<a href="#filtering-and-aggregation" class="anchor"></a>Filtering and Aggregation</h2>
<p>It is possible to filter your dataset by variance and remove genes with only 0 expression values. Also low expressed cell-types can be removed.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ds_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dataset.html">dataset</a></span><span class="op">(</span>annotation<span class="op">=</span><span class="va">annotation</span>,
                       count_matrix <span class="op">=</span> <span class="va">count_matrix</span>,
                       name<span class="op">=</span><span class="st">"filter_test"</span>, 
                       type_abundance_cutoff<span class="op">=</span><span class="fl">10</span>, 
                       filter_genes <span class="op">=</span> <span class="cn">T</span>, 
                       variance_cutoff <span class="op">=</span> <span class="fl">.1</span><span class="op">)</span></code></pre></div>
<p>Here we removed all cells which belong to a cell-type that has only 10 or less cells annotated with the <code>type_abundance_cutoff</code> parameter.<br><code>filter_genes</code> will remove all genes which have an expression value of 0 over all cells.<br>
With <code>variance_cutoff</code> you can set a cutoff to remove all genes with a variance over all cells below this value.<br></p>
<p><br></p>
<p>The pseudo-bulk sample will be generated by aggregating the expression values from all sampled cells and is then normalized. The default aggregation method is <code>sum</code>, but you also have the options <code>mean</code> and <code>median</code>. You can select them like this:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">sim_aggregation</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">ds</span>,
                                 scenario<span class="op">=</span><span class="st">"random"</span>,
                                 scaling_factor<span class="op">=</span><span class="st">"NONE"</span>,
                                 sample_aggreation <span class="op">=</span> <span class="st">"median"</span>,  <span class="co"># use median as aggregation</span>
                                 nsamples <span class="op">=</span> <span class="fl">100</span>, 
                                 ncells<span class="op">=</span><span class="fl">1000</span>,
                                 ncores<span class="op">=</span><span class="fl">4</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Fischer2020" class="csl-entry">
Fischer, David S., Leander Dony, Martin König, Abdul Moeed, Luke Zappia, Sophie Tritschler, Olle Holmberg, Hananeh Aliee, and Fabian J. Theis. 2020. <span>“Sfaira Accelerates Data and Model Reuse in Single Cell Genomics.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/2020.12.16.419036">https://doi.org/10.1101/2020.12.16.419036</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alexander Dietrich.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
