<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Inputs &amp; Outputs • SimBu</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Inputs &amp; Outputs">
<meta property="og:description" content="SimBu">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SimBu</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/simulator_documentation.html">Getting started with the pseudo-bulk simulator R package</a>
    </li>
    <li>
      <a href="../articles/simulator_input_output.html">Inputs &amp; Outputs</a>
    </li>
    <li>
      <a href="../articles/sfaira_vignette.html">Public Data Integration using Sfaira</a>
    </li>
    <li>
      <a href="../articles/simulator_scaling_factors.html">Introducing mRNA bias into simulations with scaling factors</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Inputs &amp; Outputs</h1>
                        <h4 data-toc-skip class="author">Alexander Dietrich</h4>
            
      
      
      <div class="hidden name"><code>simulator_input_output.Rmd</code></div>

    </div>

    
    
<p>This chapter covers the different input and output options of the package in detail.</p>
<div class="section level2">
<h2 id="input">Input<a class="anchor" aria-label="anchor" href="#input"></a>
</h2>
<p>The input for your simulations is always a <code>SummarizedExperiment</code> object. You can create this object with different constructing functions, which will explained below. It is also possible to merge multiple datasets objects into one.<br>
Sfaira is not covered in this vignette, but in <a href="sfaira_vignette.html">“Public Data Integration”</a>.<br></p>
<div class="section level3">
<h3 id="custom-data">Custom data<a class="anchor" aria-label="anchor" href="#custom-data"></a>
</h3>
<p>Using existing count matrices and annotations is already covered in the <a href="simulator_documentation.html">“Getting started”</a> vignette; this section will explain some minor details.<br></p>
<p>When generating a dataset with your own data, you need to provide the <code>count_matrix</code> parameter of <code><a href="../reference/dataset.html">dataset()</a></code>; additionally you can provide a TPM matrix with the <code>tpm_matrix</code>. This will then lead to two simulations, one based on counts and one based on TPMs. For either of them, genes are located in the rows, cells in the columns.<br>
Additionally, an annotation table is needed, with the cell-type annotations. It needs to consist of at least out of 2 columns: <code>ID</code> and <code>cell_type</code>, where <code>ID</code> has to be identical to the column names of the provides matrix/matrices. If not all cells appear in the annotation or matrix, the intersection of both is used to generate the dataset.<br></p>
<p>Here is some example data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/Matrix.html" class="external-link">Matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">3e5</span>, <span class="fl">5</span><span class="op">)</span>, ncol<span class="op">=</span><span class="fl">300</span><span class="op">)</span>, sparse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tpm</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/Matrix.html" class="external-link">Matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">3e5</span>, <span class="fl">5</span><span class="op">)</span>, ncol<span class="op">=</span><span class="fl">300</span><span class="op">)</span>, sparse <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">tpm</span> <span class="op">&lt;-</span> <span class="fu">Matrix</span><span class="fu">::</span><span class="fu">t</span><span class="op">(</span><span class="fl">1e6</span><span class="op">*</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu">t</span><span class="op">(</span><span class="va">tpm</span><span class="op">)</span><span class="op">/</span><span class="fu">Matrix</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">tpm</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cell-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">300</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tpm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cell-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">300</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gene-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tpm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gene-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">)</span><span class="op">)</span>

<span class="va">annotation</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="st">"ID"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cell-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">300</span><span class="op">)</span><span class="op">)</span>, 
                         <span class="st">"cell_type"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"T cells CD4"</span>,<span class="fl">50</span><span class="op">)</span>, 
                                       <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"T cells CD8"</span>,<span class="fl">50</span><span class="op">)</span>,
                                       <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Macrophages"</span>,<span class="fl">100</span><span class="op">)</span>,
                                       <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NK cells"</span>,<span class="fl">10</span><span class="op">)</span>,
                                       <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"B cells"</span>,<span class="fl">70</span><span class="op">)</span>,
                                       <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Monocytes"</span>,<span class="fl">20</span><span class="op">)</span><span class="op">)</span>,
                         row.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"cell-"</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">300</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">counts</span>, assay <span class="op">=</span> <span class="st">'counts'</span>, meta.data <span class="op">=</span> <span class="va">annotation</span><span class="op">)</span>
<span class="va">tpm_assay</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">tpm</span><span class="op">)</span>
<span class="va">seurat_obj</span><span class="op">[[</span><span class="st">'tpm'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">tpm_assay</span>

<span class="va">seurat_obj</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 2000 features across 300 samples within 2 assays </span>
<span class="co">#&gt; Active assay: counts (1000 features, 0 variable features)</span>
<span class="co">#&gt;  1 other assay present: tpm</span></code></pre></div>
<div class="section level4">
<h4 id="seurat">Seurat<a class="anchor" aria-label="anchor" href="#seurat"></a>
</h4>
<p>It is possible to use a Seurat object to build a dataset; give the name of the assay containing count data in the <code>counts</code> slot, the name of the column in the meta table with the unique cell IDs and the name of the column in the meta table with the cell type identifier. Additionally you may give the name of the assay containing TPM data in the <code>counts</code> slot.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ds_seurat</span> <span class="op">&lt;-</span> <span class="fu">SimBu</span><span class="fu">::</span><span class="fu"><a href="../reference/dataset_seurat.html">dataset_seurat</a></span><span class="op">(</span>seurat_obj <span class="op">=</span> <span class="va">seurat_obj</span>, 
                                   count_assay <span class="op">=</span> <span class="st">"counts"</span>, 
                                   cell_id_col <span class="op">=</span> <span class="st">'ID'</span>, 
                                   cell_type_col <span class="op">=</span> <span class="st">'cell_type'</span>, 
                                   tpm_assay <span class="op">=</span> <span class="st">'tpm'</span>,
                                   name <span class="op">=</span> <span class="st">"seurat_dataset"</span><span class="op">)</span>
<span class="co">#&gt; Filtering genes...</span>
<span class="co">#&gt; Created dataset.</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="h5ad-files">h5ad files<a class="anchor" aria-label="anchor" href="#h5ad-files"></a>
</h4>
<p>It is possible to use an h5ad file directly, a file format which stores <a href="https://anndata.readthedocs.io/en/latest/" class="external-link">AnnData</a> objects. As h5ad files can store cell specific information in the <code>var</code> layer, no additional annotation input for SimBu is needed.<br>
Note: if you want both counts and tpm data as input, you will have to provide two files; the annotation layer <code>var</code> has to match between these files, otherwise the intersection will be used. You can provide the column names in <code>var</code> which store cell id and cell type with the <code>cell_id_col</code> and <code>cell_type_col</code> parameters respectively.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ad</span> <span class="op">&lt;-</span> <span class="fu">anndata</span><span class="fu">::</span><span class="fu"><a href="https://anndata.dynverse.org/reference/AnnData.html" class="external-link">AnnData</a></span><span class="op">(</span>X<span class="op">=</span><span class="va">counts</span>, 
                       obs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>genes<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, row.names<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">)</span>, <span class="co"># obs stores gene information</span>
                       var <span class="op">=</span> <span class="va">annotation</span><span class="op">)</span>                                                     <span class="co"># var stores cell information</span>
<span class="va">counts_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span>pattern <span class="op">=</span> <span class="st">'test'</span>,fileext <span class="op">=</span> <span class="st">'.h5ad'</span><span class="op">)</span>
<span class="fu">anndata</span><span class="fu">::</span><span class="fu"><a href="https://anndata.dynverse.org/reference/write_h5ad.html" class="external-link">write_h5ad</a></span><span class="op">(</span>anndata <span class="op">=</span> <span class="va">ad</span>, filename <span class="op">=</span> <span class="va">counts_file</span><span class="op">)</span>


<span class="va">ds_h5ad</span> <span class="op">&lt;-</span> <span class="fu">SimBu</span><span class="fu">::</span><span class="fu"><a href="../reference/dataset_h5ad.html">dataset_h5ad</a></span><span class="op">(</span>h5ad_file_counts <span class="op">=</span> <span class="va">counts_file</span>,
                               name <span class="op">=</span> <span class="st">"h5ad_dataset"</span>,
                               cell_id_col <span class="op">=</span> <span class="st">'ID'</span>, 
                               cell_type_col <span class="op">=</span> <span class="st">'cell_type'</span><span class="op">)</span>
<span class="co">#&gt; Filtering genes...</span>
<span class="co">#&gt; Created dataset.</span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="merging-datasets">Merging datasets<a class="anchor" aria-label="anchor" href="#merging-datasets"></a>
</h3>
<p>You are able to merge multiple datasets by using the <code>dataset_merge</code> function:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">ds</span> <span class="op">&lt;-</span> <span class="fu">SimBu</span><span class="fu">::</span><span class="fu"><a href="../reference/dataset.html">dataset</a></span><span class="op">(</span>annotation <span class="op">=</span> <span class="va">annotation</span>,
                     count_matrix <span class="op">=</span> <span class="va">counts</span>,
                     tpm_matrix <span class="op">=</span> <span class="va">tpm</span>,
                     name <span class="op">=</span> <span class="st">"test_dataset"</span><span class="op">)</span>
<span class="co">#&gt; Filtering genes...</span>
<span class="co">#&gt; Created dataset.</span>

<span class="va">ds_multiple</span> <span class="op">&lt;-</span> <span class="fu">SimBu</span><span class="fu">::</span><span class="fu"><a href="../reference/dataset_merge.html">dataset_merge</a></span><span class="op">(</span>dataset_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">ds_seurat</span>, <span class="va">ds</span><span class="op">)</span>,
                                    name <span class="op">=</span> <span class="st">"ds_multiple"</span><span class="op">)</span>
<span class="co">#&gt; Filtering genes...</span>
<span class="co">#&gt; Created dataset.</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h2>
<p>The <code>simulation</code> object contains three named entries:<br></p>
<ul>
<li>
<code>bulk</code>: a SummarizedExperiment object with the pseudo-bulk dataset(s) stored in the <code>assays</code>. They can be accessed like this:</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">simulation</span> <span class="op">&lt;-</span> <span class="fu">SimBu</span><span class="fu">::</span><span class="fu"><a href="../reference/simulate_bulk.html">simulate_bulk</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">ds_multiple</span>,
                                   scenario <span class="op">=</span> <span class="st">"random"</span>, 
                                   scaling_factor <span class="op">=</span> <span class="st">"NONE"</span>,
                                   nsamples <span class="op">=</span> <span class="fl">10</span>, ncells <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="co">#&gt; Finished simulation.</span>

<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">simulation</span><span class="op">$</span><span class="va">bulk</span><span class="op">)</span><span class="op">[[</span><span class="st">"bulk_counts"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] 1000   10</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">simulation</span><span class="op">$</span><span class="va">bulk</span><span class="op">)</span><span class="op">[[</span><span class="st">"bulk_tpm"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] 1000   10</span></code></pre></div>
<p>If only the count matrix was given to the dataset initially, only the <code>bulk_counts</code> assay is filled.</p>
<ul>
<li><p><code>cell_fractions</code>: a table where rows represent the simulated samples and columns represent the different simulated cell-types. The entries in the table store the specific cell-type fraction per sample.<br></p></li>
<li><p><code>scaling_vector</code>: a named list, with the used scaling value for each cell from the single cell dataset.<br></p></li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.0 (2021-05-18)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 18.04.6 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] systemfonts_1.0.4           plyr_1.8.7                 </span>
<span class="co">#&gt;   [3] igraph_1.2.11               lazyeval_0.2.2             </span>
<span class="co">#&gt;   [5] proxyC_0.2.4                splines_4.1.0              </span>
<span class="co">#&gt;   [7] BiocParallel_1.26.2         listenv_0.8.0              </span>
<span class="co">#&gt;   [9] scattermore_0.8             GenomeInfoDb_1.28.4        </span>
<span class="co">#&gt;  [11] ggplot2_3.3.5               digest_0.6.27              </span>
<span class="co">#&gt;  [13] htmltools_0.5.2             fansi_1.0.3                </span>
<span class="co">#&gt;  [15] magrittr_2.0.1              memoise_2.0.1              </span>
<span class="co">#&gt;  [17] tensor_1.5                  cluster_2.1.2              </span>
<span class="co">#&gt;  [19] ROCR_1.0-11                 globals_0.14.0             </span>
<span class="co">#&gt;  [21] RcppParallel_5.1.5          matrixStats_0.61.0         </span>
<span class="co">#&gt;  [23] pkgdown_2.0.2               spatstat.sparse_2.1-0      </span>
<span class="co">#&gt;  [25] colorspace_2.0-3            rappdirs_0.3.3             </span>
<span class="co">#&gt;  [27] ggrepel_0.9.1               textshaping_0.3.6          </span>
<span class="co">#&gt;  [29] xfun_0.30                   dplyr_1.0.8                </span>
<span class="co">#&gt;  [31] RCurl_1.98-1.6              crayon_1.5.1               </span>
<span class="co">#&gt;  [33] jsonlite_1.7.2              spatstat.data_2.1-2        </span>
<span class="co">#&gt;  [35] survival_3.2-11             zoo_1.8-9                  </span>
<span class="co">#&gt;  [37] glue_1.6.2                  polyclip_1.10-0            </span>
<span class="co">#&gt;  [39] gtable_0.3.0                zlibbioc_1.38.0            </span>
<span class="co">#&gt;  [41] XVector_0.32.0              SimBu_0.1.0                </span>
<span class="co">#&gt;  [43] leiden_0.3.9                DelayedArray_0.18.0        </span>
<span class="co">#&gt;  [45] future.apply_1.8.1          BiocGenerics_0.38.0        </span>
<span class="co">#&gt;  [47] abind_1.4-5                 scales_1.1.1               </span>
<span class="co">#&gt;  [49] DBI_1.1.2                   spatstat.random_2.1-0      </span>
<span class="co">#&gt;  [51] miniUI_0.1.1.1              Rcpp_1.0.8.3               </span>
<span class="co">#&gt;  [53] viridisLite_0.4.0           xtable_1.8-4               </span>
<span class="co">#&gt;  [55] reticulate_1.24             spatstat.core_2.4-0        </span>
<span class="co">#&gt;  [57] stats4_4.1.0                htmlwidgets_1.5.4          </span>
<span class="co">#&gt;  [59] httr_1.4.2                  anndata_0.7.5.3            </span>
<span class="co">#&gt;  [61] RColorBrewer_1.1-2          ellipsis_0.3.2             </span>
<span class="co">#&gt;  [63] Seurat_4.1.0                ica_1.0-2                  </span>
<span class="co">#&gt;  [65] pkgconfig_2.0.3             sass_0.4.1                 </span>
<span class="co">#&gt;  [67] uwot_0.1.11                 deldir_1.0-6               </span>
<span class="co">#&gt;  [69] here_1.0.1                  utf8_1.2.2                 </span>
<span class="co">#&gt;  [71] tidyselect_1.1.2            rlang_1.0.2                </span>
<span class="co">#&gt;  [73] reshape2_1.4.4              later_1.3.0                </span>
<span class="co">#&gt;  [75] munsell_0.5.0               tools_4.1.0                </span>
<span class="co">#&gt;  [77] cachem_1.0.6                cli_3.2.0                  </span>
<span class="co">#&gt;  [79] generics_0.1.2              ggridges_0.5.3             </span>
<span class="co">#&gt;  [81] evaluate_0.14               stringr_1.4.0              </span>
<span class="co">#&gt;  [83] fastmap_1.1.0               yaml_2.3.5                 </span>
<span class="co">#&gt;  [85] ragg_1.2.2                  goftest_1.2-3              </span>
<span class="co">#&gt;  [87] knitr_1.33                  fs_1.5.2                   </span>
<span class="co">#&gt;  [89] fitdistrplus_1.1-8          purrr_0.3.4                </span>
<span class="co">#&gt;  [91] RANN_2.6.1                  pbapply_1.5-0              </span>
<span class="co">#&gt;  [93] future_1.24.0               nlme_3.1-152               </span>
<span class="co">#&gt;  [95] sparseMatrixStats_1.4.2     mime_0.10                  </span>
<span class="co">#&gt;  [97] compiler_4.1.0              rstudioapi_0.13            </span>
<span class="co">#&gt;  [99] plotly_4.10.0               png_0.1-7                  </span>
<span class="co">#&gt; [101] spatstat.utils_2.3-0        tibble_3.1.6               </span>
<span class="co">#&gt; [103] bslib_0.3.1                 stringi_1.6.1              </span>
<span class="co">#&gt; [105] desc_1.4.1                  lattice_0.20-44            </span>
<span class="co">#&gt; [107] Matrix_1.3-4                vctrs_0.3.8                </span>
<span class="co">#&gt; [109] pillar_1.7.0                lifecycle_1.0.1            </span>
<span class="co">#&gt; [111] spatstat.geom_2.3-2         lmtest_0.9-40              </span>
<span class="co">#&gt; [113] jquerylib_0.1.4             RcppAnnoy_0.0.19           </span>
<span class="co">#&gt; [115] bitops_1.0-7                data.table_1.14.2          </span>
<span class="co">#&gt; [117] cowplot_1.1.1               irlba_2.3.5                </span>
<span class="co">#&gt; [119] GenomicRanges_1.44.0        httpuv_1.6.5               </span>
<span class="co">#&gt; [121] patchwork_1.1.1             R6_2.5.1                   </span>
<span class="co">#&gt; [123] promises_1.2.0.1            KernSmooth_2.23-20         </span>
<span class="co">#&gt; [125] gridExtra_2.3               IRanges_2.26.0             </span>
<span class="co">#&gt; [127] parallelly_1.30.0           codetools_0.2-18           </span>
<span class="co">#&gt; [129] MASS_7.3-54                 assertthat_0.2.1           </span>
<span class="co">#&gt; [131] SummarizedExperiment_1.22.0 rprojroot_2.0.2            </span>
<span class="co">#&gt; [133] SeuratObject_4.0.4          sctransform_0.3.3          </span>
<span class="co">#&gt; [135] GenomeInfoDbData_1.2.6      S4Vectors_0.30.2           </span>
<span class="co">#&gt; [137] mgcv_1.8-36                 parallel_4.1.0             </span>
<span class="co">#&gt; [139] grid_4.1.0                  rpart_4.1-15               </span>
<span class="co">#&gt; [141] tidyr_1.2.0                 rmarkdown_2.13             </span>
<span class="co">#&gt; [143] MatrixGenerics_1.4.3        Rtsne_0.15                 </span>
<span class="co">#&gt; [145] Biobase_2.52.0              shiny_1.7.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Alexander Dietrich.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
